src/gui/plugins/component_inspector/ComponentInspector.qml

```diff
@@ -227,6 +247,100 @@ Rectangle {
     }
   }
 
+  Dialog {
+    id: addSystemDialog
+    modal: false
+    focus: true
+    title: "Add System"
+    closePolicy: Popup.CloseOnEscape
+
+    ColumnLayout {
+      GridLayout {
+        columns: 2
+        columnSpacing: 30
+        Text {
+          text: "Name"
+          Layout.row: 0
+          Layout.column: 0
+        }
+
+        TextField {
+          id: nameField
+          selectByMouse: true
+          Layout.row: 0
+          Layout.column: 1
+          Layout.minimumWidth: 250
+          onTextEdited: {
+            addSystemDialog.updateButtonState();
+          }
+        }
+
+        Text {
+          text: "Filename"
```

Comment:
So one usability issue with the current dialog is that the user needs to know the plugin filename and name to insert a plugin. Most beginners won't even know where to look for this.

We could try an approach similar to the GUI plugin menu, where we crawl all possible plugin sources (default directories, env vars...) and list all plugins to the user. We'll probably need to add a service from `SystemLoader` to provide this list to the GUI.

The other tricky part with the server plugins is that we can easily get the `Filename`, but the `Name` isn't so easy. I think one workaround would be to load the first plugin in the library when no name is provided here:

https://github.com/gazebosim/gz-sim/blob/6739800821af5e39767fe22e8da4b3e27fbaa284/src/SystemLoader.cc#L79-L88

I think this would increase the scope of this PR by a lot, so I'm ok leaving it for a follow-up PR. What do you think?

Reply:
yeah ok that sounds good. We can do this in a follow-up PR.

Reply:
Done in #1581